scripts/Diffraction/isis_powder/routines/run_details.py

```diff
@@ -67,6 +68,10 @@ def create_run_details_object(
     # Sample empty if there is one as this is instrument specific
     sample_empty = getattr(inst_settings, "sample_empty", None)
 
+    # Override sample empty run if it's passed as a parameter
+    if sample_empty_run:
+        sample_empty = sample_empty_run
```

Comment:
It feels like we should be able to arrange the code so this isn't necessary. Can you just do:
```
inst_settings.sample_empty = empty_can_runs
```
in osiris_algs.py at around ~L272 before the call to create_run_details_object?

Reply:
Done